fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/OCFLPersistentStorageSession.java

```diff
@@ -153,12 +189,37 @@ public InputStream getBinaryContent(final String identifier, final Instant versi
     }
 
     @Override
-    public void commit() throws PersistentStorageException {
+    public void commit(final CommitOption option) throws PersistentStorageException {
         if (isReadOnly()) {
             // No changes to commit.
             return;
         }
-        // commit changes.
+
+        try {
+            mutex.lock();
+            //prepare session for commit
+            final Collection<OCFLObjectSession> sessions = this.sessionMap.values();
+
+            for(OCFLObjectSession objectSession : sessions){
+                objectSession.prepare();
+            }
+
+            //perform commit
+            for(OCFLObjectSession objectSession : sessions){
+                objectSession.commit(option);
+            }
+
+            //close each session
+            for(OCFLObjectSession objectSession : sessions){
+                objectSession.close();
+            }
+
+            //purge object sessions
+            sessionMap.clear();
```

Comment:
I wasn't sure about session reuse in the case of the OCFLPeristentStorageSession.  I guess we were thinking it would be associated with a transaction so I suppose not.